src/shogun/regression/LinearRidgeRegression.cpp

```diff
@@ -32,8 +32,66 @@ CLinearRidgeRegression::CLinearRidgeRegression(float64_t tau, CDenseFeatures<flo
 void CLinearRidgeRegression::init()
 {
 	set_tau(1e-6);
+	m_use_bias = true;
 
 	SG_ADD(&m_tau, "tau", "Regularization parameter", MS_AVAILABLE);
+	SG_ADD(
+	    &m_use_bias, "use_bias", "Whether or not to fit an offset term",
+	    MS_NOT_AVAILABLE);
+}
+
+template <typename T>
+bool CLinearRidgeRegression::train_machine_templated(
+    const CDenseFeatures<T>* feats)
+{
+	auto N = feats->get_num_vectors();
+	auto D = feats->get_num_features();
+
+	auto y = regression_labels(m_labels)->get_labels();
+	SGVector<T> x_bar;
+	T y_bar;
+	if (m_use_bias)
+	{
+		x_bar = feats->mean();
+		y_bar = linalg::mean(y);
+	}
+
+	SGVector<float64_t> w;
+	if (N >= D)
+	{
+		SGMatrix<T> cov = feats->cov();
+		linalg::add_ridge(cov, m_tau);
+		auto L = linalg::cholesky_factor(cov);
+
+		if (m_use_bias)
```

Comment:
@iglesias voila! :)
If you could do the math and confirm this, that would be neat! :)